apps/desktop-wallet/src/utils/transactions.ts

```diff
@@ -124,73 +129,166 @@ export const directionOptions: {
   addressHash: AddressHash,
-  showInternalInflows?: boolean
-): TransactionInfo => {
+  isInAddressDetailsModal?: boolean
+): TransactionInfoType => {
   const state = store.getState()
-  const fungibleTokens = state.fungibleTokens.entities
   const internalAddresses = state.addresses.ids as AddressHash[]
 
-  let amount: bigint | undefined = BigInt(0)
-  let direction: TransactionDirection
-  let infoType: TransactionInfoType
-  let lockTime: Date | undefined
-  let tokens: Required<AssetAmount>[] = []
-
   if (isPendingTx(tx)) {
-    direction = internalAddresses.includes(tx.toAddress) ? (addressHash === tx.fromAddress ? 'out' : 'in') : 'out'
-    infoType = 'pending'
-    amount = tx.amount ? convertToNegative(BigInt(tx.amount)) : undefined
-    tokens = tx.tokens ? tx.tokens.map((token) => ({ ...token, amount: convertToNegative(BigInt(token.amount)) })) : []
-    lockTime = tx.lockTime !== undefined ? new Date(tx.lockTime) : undefined
+    return 'pending'
+  } else if (isConsolidationTx(tx)) {
+    return 'move'
   } else {
-    const { alph: alphAmount, tokens: tokenAmounts } = calcTxAmountsDeltaForAddress(tx, addressHash)
+    const { alphAmount, tokenAmounts } = calcTxAmountsDeltaForAddress(tx, addressHash)
 
-    amount = alphAmount
-    tokens = tokenAmounts.map((token) => ({ ...token, amount: token.amount }))
-
-    if (isConsolidationTx(tx)) {
-      direction = 'out'
-      infoType = 'move'
-    } else if (isSwap(amount, tokens)) {
-      direction = 'swap'
-      infoType = 'swap'
+    if (hasPositiveAndNegativeAmounts(alphAmount, tokenAmounts)) {
+      return 'swap'
     } else {
-      direction = getDirection(tx, addressHash)
       const isInternalTransfer = isInternalTx(tx, internalAddresses)
-      infoType =
-        (isInternalTransfer && showInternalInflows && direction === 'out') ||
-        (isInternalTransfer && !showInternalInflows)
-          ? 'move'
-          : direction
-    }
+      const alphAmountReduced = alphAmount < 0 // tokenAmounts is checked in the swap condition
 
-    lockTime = (tx.outputs ?? []).reduce(
-      (a, b) =>
-        a > new Date((b as explorer.AssetOutput).lockTime ?? 0)
-          ? a
-          : new Date((b as explorer.AssetOutput).lockTime ?? 0),
-      new Date(0)
-    )
-    lockTime = lockTime.toISOString() === new Date(0).toISOString() ? undefined : lockTime
+      if (
+        (isInternalTransfer && isInAddressDetailsModal && alphAmountReduced) ||
+        (isInternalTransfer && !isInAddressDetailsModal)
+      ) {
+        return 'move'
+      } else {
+        if (alphAmountReduced) {
+          return 'out'
+        } else {
+          return 'in'
+        }
+      }
+    }
   }
+}
 
-  const tokenAssets = [...tokens.map((token) => ({ ...token, ...fungibleTokens[token.id] }))]
-  const assets = amount !== undefined ? [{ ...ALPH, amount }, ...tokenAssets] : tokenAssets
+export const useTransactionLockTime = (tx: Transaction | PendingTransaction) =>
```

Comment:
I feel like many of the hooks define in this file could be moved to the `transactionsDisplay` folder.. WDYT?

Reply:
You're probably right, I'll have a look!

Reply:
Good one! I split most of these functions and hooks into `features/transactionsDisplay` and `features/send` in this commit: e195942bd

Only a very small file remains: https://github.com/alephium/alephium-frontend/blob/next-tanstack-address-token-rows/apps/desktop-wallet/src/utils/transactions.ts